File: 2211-count-collisions-on-a-road/2211-count-collisions-on-a-road.java
class Solution {
    public int countCollisions(String directions) {
       char [] arr = directions.toCharArray();
       int n = arr.length;
       int i=0;
       int ans=0;
       int j = n-1;
       while(i<n && arr[i]=='L') i++;
       while(j>=0 && arr[j]=='R') j--;
       for(int k=i; k<=j; k++){
        if(arr[k]!='S') ans++;
       }
       return ans;
    }
}

File: 2211-count-collisions-on-a-road/2211-count-collisions-on-a-road.py
class Solution:
    def countCollisions(self, directions: str) -> int:
        arr = list(directions)
        n = len(arr)
        i=0
        j=n-1
        ans =0
        while i<n and arr[i]=='L': i+=1
        while j>=0 and arr[j]=='R': j-=1
        for k in range(i, j+1):
            if arr[k] != 'S':
                ans +=1
        
        return ans